core/Plugin/Archiver.php

```diff
@@ -83,6 +87,92 @@ public function __construct(ArchiveProcessor $processor)
         $this->enabled = true;
     }
 
+    private function getPluginName()
+    {
+        $className = get_class($this);
+        $parts = explode('\\', $className);
+        $parts = array_filter($parts);
+        $plugin = $parts[2];
+
+        if (!Manager::getInstance()->isPluginLoaded($plugin)) {
+            return null;
+        }
```

Comment:
I get what this does, but it doesn't feel clean. The method has a side effect of acting as `isPluginLoaded` which can be added and used instead further down in the code.